crates/client/src/rr/dnssec/tsig.rs

```diff
@@ -136,12 +141,14 @@ impl TSigner {
         // this is technically in violation of the RFC, in case both time and
         // truncation policy are bad, time should be reported and this code will report
         // truncation issue instead
+
```

Comment:
@trinity-1686a while working on this I had a little more time to consider the verification steps I was noticing that we're not using a constant time function for the signature comparison, which can lead to timing attacks. I was curious how important the ability to verify a truncated mac is? I know it's to the RFC's spec, but I wonder if that's outdated?

Reply:
I tried to be on-spec, but when debugging (against Knot), I don't think I ever received a truncated MAC. If you're not sure if it should be supported, I think what you could do is detect if len does not match, and return an error message asking to fill an issue. If anyone really needs it, then they can ask for it be supported.